modules/host/contractmanager/sectorremoval.go

```diff
@@ -0,0 +1,233 @@
+package contractmanager
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+	"time"
+
+	"gitlab.com/NebulousLabs/errors"
+	"go.sia.tech/siad/crypto"
+	"go.sia.tech/siad/modules"
+)
+
+const removalEntrySize = 20
+
+type (
+	removalEntry struct {
+		ID sectorID
+		// Count refers to the number of references that should be removed from
+		// the contract manager.
+		Count uint64
+		// Offset refers to the position in the persistence file of the entry
+		Offset int64
+	}
+
+	sectorRemovalMap struct {
+		mu          sync.Mutex
+		sectors     map[sectorID]removalEntry
+		entries     int64
+		removalFile *os.File
+
+		cm  *ContractManager
+		wal *writeAheadLog
+	}
+)
+
+// writes the removal entry to disk
+func (sq *sectorRemovalMap) writeRemovalEntry(id sectorID, count uint64, offset int64) error {
+	buf := make([]byte, removalEntrySize)
+	copy(buf[:], id[:])
+	binary.LittleEndian.PutUint64(buf[12:], count)
+	_, err := sq.removalFile.WriteAt(buf, offset)
+	return err
+}
+
+// managedLoad loads the removal queue from disk. Entries with a count of zero
+// and read errors are ignored. The file is compacted while loading.
+func (sq *sectorRemovalMap) managedLoad(path string) (err error) {
+	sq.mu.Lock()
+	defer sq.mu.Unlock()
+
+	sq.removalFile, err = os.OpenFile(path, os.O_RDWR|os.O_CREATE, modules.DefaultFilePerm)
+	if err != nil {
+		return errors.AddContext(err, "unable to open sector removal queue file")
+	}
+	if _, err := sq.removalFile.Seek(0, 0); err != nil {
+		return errors.AddContext(err, "unable to seek to beginning of sector removal queue file")
+	}
+	buf := make([]byte, 20)
+	// read each entry from disk. The persistence file is compacted while
+	// reading.
+	for i := int64(0); ; i++ {
+		if _, err := sq.removalFile.ReadAt(buf, i*removalEntrySize); err == io.EOF || err == io.ErrUnexpectedEOF {
```

Comment:
`ReadAt` implies we're reading out of order, but it looks like this is reading the whole file sequentially. Consider `io.ReadFull` instead.

A `bufio.Reader` seems warranted as well, since `removalEntrySize` is quite small.